demo/full/scripts/controllers/ContentList.jsx

```diff
@@ -240,21 +426,52 @@ class ContentList extends React.Component {
         />);
     };
 
+    const isLocalContent = !!(chosenContent && chosenContent.localContent);
+    const chosenContentHasDRMInfo = chosenContent &&
+                        chosenContent.drmInfos &&
+                        chosenContent.drmInfos[0];
+
     return (
       <div className="choice-inputs-wrapper">
         <div className="content-inputs">
-          <div className="content-inputs-selects">
-            <Select
-              className="choice-input transport-type-choice white-select"
-              onChange={onTechChange}
-              options={TRANSPORT_TYPES}
-            />
-            <Select
-              className="choice-input content-choice white-select"
-              onChange={onContentChange}
-              options={contentsToSelect}
-              selected={contentChoiceIndex}
-            />
+          <div className="content-inputs-left">
+            <div className="content-inputs-selects">
+              <Select
+                className="choice-input transport-type-choice white-select"
+                onChange={onTechChange}
+                options={TRANSPORT_TYPES}
+              />
+              <Select
+                className="choice-input content-choice white-select"
+                onChange={onContentInputChange}
+                options={contentsToSelect}
+                selected={contentChoiceIndex}
+              />
+            </div>
+            <div className="content-inputs-middle">
+              {
+                (hasTextInput || isLocalContent) ?
+                  (<Button
+                    className={"choice-input-button save-button" +
+                      (!activeSaveOption ? " disabled" : "")}
+                    onClick={onClickSave}
+                    disabled={!activeSaveOption || isSavingOrUpdating}
+                    value={isLocalContent ?
+                      (isSavingOrUpdating ? "Updating" : "Update content") :
+                      (isSavingOrUpdating ? "Saving" : "Save content")}
+                  />) :
+                  null
+              }
+              {
+                isLocalContent ?
```

Comment:
Why two times in a row?
I don't remember if it became possible to chain multiple elements without a parent in JSX but even if that's not the case it may be cleaner to just add a parent.